### PR TITLE
Species lock for ghost spawners

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

You can now species lock a ghost spawner.

Makes HECU and Enclave human only.

Ashwalker and pod person spawns are also locked to the native species.

## Changelog

:cl:
add: Some spawners now have species restrictions.
/:cl:

### CODE REVIEW

File: code/modules/mob_spawn/mob_spawn.dm [hunk 1]
```diff
@@ -156,6 +158,11 @@
 /obj/effect/mob_spawn/ghost_role/attack_ghost(mob/user)
 	if(!SSticker.HasRoundStarted() || !loc)
 		return
+	// SKYRAT EDIT ADDITION
+	if(restricted_species && user.client?.prefs?.read_preference(/datum/preference/choiced/species) != restricted_species)
```
Comment (jjpark-kb) [inline code/modules/mob_spawn/mob_spawn.dm]: I feel it should instead ask if its a type within the var... just so you can have multiple species and it isn't locked to a singular species.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -303,10 +303,6 @@ GLOBAL_LIST_EMPTY(customizable_races)
 	default_mutant_bodyparts = list("tail" = "Cat", "ears" = "Cat")
 	learnable_languages = list(/datum/language/common, /datum/language/nekomimetic)
 
-/datum/species/human/monkey
-	mutant_bodyparts = list()
-	default_mutant_bodyparts = list("tail" = "Monkey")
```
Comment (jjpark-kb) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm:308]: what is this for?

Comment (Gandalf2k15) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species.dm:308]: Redundant and overrode human species for some reason

File: code/modules/mob_spawn/mob_spawn.dm (hunk 1 above)
Comment (Gandalf2k15) [reply code/modules/mob_spawn/mob_spawn.dm]: Hmm, sure

File: modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa.dm [hunk 3]
```diff
@@ -605,6 +605,7 @@
 	prompt_name = "a tactical squad member"
 	outfit = /datum/outfit/hecu
 	you_are_text = "You are an elite tactical squad deployed into the research facility to contain the infestation. DO NOT TRY TO EXPLORE THE LEVEL. STAY AROUND YOUR AREA."
+	restricted_species = /datum/species/human
```
Comment (Zonespace27) [inline modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa.dm]: mind applying this to the primal plantperson and the ashwalker spawns?

Comment (Gandalf2k15) [reply modular_skyrat/master_files/code/modules/awaymission/mission_code/black_mesa.dm]: sure

**Pull request merged**